app/controllers/source_sets_controller.rb

```diff
@@ -67,4 +67,11 @@ def source_set_params
                                        author_ids: [],
                                        tag_ids: [])
   end
+
+  ##
+  # @return [Array<Tag>]
+  def get_tags_from_params
+    return nil unless params[:tags].present?
+    params[:tags].map { |slug| Tag.find_by_slug(slug) }.compact
```

Comment:
One issue with my suggestion is it might require some input sanitization if that's not done automatically.